lib/licensed/source/pip.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+require "json"
+require "English"
+
+module Licensed
+  module Source
+    class Pip
+      def initialize(config)
+        @config = config
+      end
+
+      def type
+        "python"
+      end
+
+      def enabled?
+        @config.enabled?(type) && File.exist?(@config.pwd.join("requirements.txt"))
+      end
+
+      def dependencies
+        packages = parse_requirements_txt
+
+        @dependencies = packages.map do |package_name|
+            package = package_info(package_name)
+            location = File.join(package["Location"], "-" + package["Version"] + ".dist-info")
+            Dependency.new(location, {
+              "type"        => type,
+              "name"        => package["Name"],
+              "summary"     => package["Summary"],
+              "homepage"    => package["Home-page"],
+              "version"     => package["Version"]
+            })
+        end
+      end
+
+      # Build the list of packages from a 'requirements.txt'
+      # Assumes that the requirements.txt follow the format pkg=1.0.0 or pkg==1.0.0
+      def parse_requirements_txt
+        packages = []
+        File.open(@config.pwd.join("requirements.txt")).each do |line|
+          p_split = line.split("=")
+          packages.push(p_split[0])
+        end
+        packages
+      end
+
+      def package_info(package_name)
+        info = {}
+        p_info = pip_command(package_name).split("\n")
+        p_info.each do |pkg|
+          k, v = pkg.split(":", 2)
+          info[k&.strip] = v&.strip
```

Comment:
we probably don't want any nil or empty strings as keys.  can you add a `next if k.nil? || k.empty?`

Reply:
are the calls to `&.strip` due to the lines containing "\n" line endings?  if so it would make more sense to do `pip_command(package_name).split("\n").map(:&strip)` to remove them before evaluating the string content

Reply:
`strip` because some of the packages have whitespace when parsing the output of the shelled out command, which messes up the `location` building.